meshmode/mesh/processing.py

```diff
@@ -173,31 +174,36 @@ def _filter_mesh_groups(mesh, selected_elements, vertex_id_dtype):
 
     from dataclasses import replace
     new_groups = [
-            replace(mesh.groups[i_old_grp],
-                vertex_indices=new_vertex_indices[i_new_grp],
-                nodes=mesh.groups[i_old_grp].nodes[
-                    :, filtered_group_elements[i_new_grp], :].copy(),
+            replace(grp,
+                vertex_indices=new_vertex_indices[igrp],
+                nodes=grp.nodes[:, filtered_group_elements[igrp], :].copy(),
                 element_nr_base=None, node_nr_base=None)
-            for i_new_grp, i_old_grp in enumerate(new_group_to_old_group)]
+            for igrp, grp in enumerate(mesh.groups)]
 
-    return new_groups, group_to_new_group, required_vertex_indices
+    return new_groups, required_vertex_indices
 
 
-def _get_connected_partitions(
-        mesh, part_per_element, global_elem_to_part_elem):
+def _get_connected_parts(
+        mesh: Mesh,
+        part_id_to_part_index: Dict[PartID, int],
```

Comment:
```suggestion
        part_id_to_part_index: Mapping[PartID, int],
```
since you're not writing it. (Here and elsewhere.)